### ISSUE TITLE
Plurals Considered Harmful

### ISSUE THREAD

**Issue by zoffyzhang** [kettanaito/naming-cheatsheet#38]
I think plurals are side effects of English language, because plurals are noun mix up with quantifier, regardless of the complexity of plural spell rules, sometimes plurals are redundant and inappropriate in coding language.

so my suggestion is to use **Array** or **List** as quantifier rather than plurals.

```js
/* Bad */
const friends = ['Bob']
const friends = [ ]

/* Good */
const friendArray = ['Bob']
const friendArray = [ ]

/* Good as well */
const friendList = ['Bob']
const friendList = [ ]
```

**Comment by octavian-nita** [comment]
Wouldn't that unnecessarily bring an implementation detail into the name? What about functions/routines? fetch friendliest()?

**Comment by zoffyzhang** [comment]
1. I think Array and List are only type definitions, not heavy as implementation detail 
2. I do use function name like `getRecordList`

**Comment by khang-nd** [comment]
There's already a closed issue (#24) discussing this.

**Comment by kettanaito** [comment]
Hey, @zoffyzhang. Thanks for suggesting this. 

I think your approach of adding Array and List to the names is a perfectly valid choice. I wouldn't however state that plurals are necessarily harmful. I see `const friends = []` as an equally good variable name. I tend to lean forward that a variable name represents the data that it stores more than the way it stores it. 

For example, consider this:

```js
const friends = ['Bob', 'Rita']
```

Perhaps you migrate that `friends` array to a `Set` one day:

```js
const friends = new Set(['Bob', 'Rita'])
```

I like how you preserve the variable name and don't have to change from `friendsList` to `friendsSet`, because the data the variable holds hasn't changed. 

We should be extremely cautious when discouraging anything as a part of these guidelines. I view this document as a good encouragement with examples and expect people to build on top of those encouragement (just as you can add `xArray`!).
